alternator/expression_factory.cc

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "expressions.hh"
+#include "utils/log.hh"
+
+#include <boost/intrusive/list.hpp>
+#include <absl/container/node_hash_map.h>
+#include <optional>
+
+logging::logger cache_logger("alternator-cache");
+
+namespace alternator {
+namespace {
+
+template<typename Value>
+class lru_string_map {
+    struct key {
+        mutable boost::intrusive::list_member_hook<> list_hook;
+        seastar::sstring str;
```

Comment:
In Alternator we usually use std::string, not seastar::sstring. It made it easier to use rapidjson (see https://github.com/scylladb/seastar/issues/634), and this has become the norm in Alternator unless you're interoperating with the rest of Scylla.